icu4c/source/data/misc/units.txt

```diff
@@ -1,54 +1,321 @@
+        }
+        gallon{
+            factor{"cup2m3*16"}
+            target{"cubic-meter"}
+        }
+        gallon-imperial{
+            factor{"0.00454609"}
+            target{"cubic-meter"}
+        }
+        gram{
+            factor{"0.001"}
+            target{"kilogram"}
+        }
+        hectare{
+            factor{"10000"}
+            target{"square-meter"}
+        }
+        hertz{
+            factor{"1"}
+            target{"one-per-second"}
+        }
+        horsepower{
+            factor{"ft2m * lb2kg * gravity * 550"}
+            target{"square-meter-kilogram-per-cubic-second"}
+        }
+        hour{
+            factor{"3600"}
+            target{"second"}
+        }
+        inch{
+            factor{"f2m/12"}
+            target{"meter"}
+        }
+        inch-hg{
+            factor{"3386.389"}
+            target{"kilogram-per-meter-square-second"}
+        }
+        joule{
+            factor{"1"}
+            target{"kilogram-square-meter-per-square-second"}
+        }
+        karat{
+            factor{"1/24"}
+            target{"one-per-one"}
+        }
         kelvin{
             factor{"1"}
             offset{"273.15"}
             target{"celsius"}
         }
-        // <convertUnit source='hectare' target='square-meter' factor='10000'/>
-        // <convertUnit source='acre' target='square-meter' factor='f2m^2 * 43560'/>
-        // <convertUnit source='dunam' target='square-meter' factor='1000'/>
-        hectare{
-            factor{"10000"}
-            target{"square-meter"}
+        knot{
+            factor{"1852/3600"}
+            target{"meter-per-second"}
         }
-        acre{
-            factor{"f2m^2 * 43560"}
-            target{"square-meter"}
+        light-year{
+            factor{"9460730000000000"}
+            target{"meter"}
         }
-        dunam{
+        liter{
+            factor{"0.001"}
+            target{"cubic-meter"}
+        }
+        metric-ton{
             factor{"1000"}
-            target{"square-meter"}
+            target{"kilogram"}
+        }
+        mile{
+            factor{"f2m*5280"}
+            target{"meter"}
+        }
+        mile-per-gallon{
+            factor{"235.215"}
+            reciprocal{"TRUE"}
+            target{"liter-per-kilometer"}
+        }
+        mile-scandinavian{
+            factor{"10000"}
+            target{"meter"}
+        }
+        minute{
+            factor{"3600"}
+            target{"second"}
+        }
+        mole{
+            factor{"6.02214076E+23"}
+            target{"one"}
+        }
+        nautical-mile{
+            factor{"1852"}
+            target{"meter"}
+        }
+        newton{
+            factor{"1"}
+            target{"kilogram-meter-per-square-second"}
+        }
+        ohm{
+            factor{"1"}
+            target{"kilogram-square-meter-per-cubic-second-square-ampere"}
+        }
+        one{
+            factor{"1"}
+            target{"1"}
```

Comment:
We should make the target `"one"`, not `"1"`, but that's a bug in the XML data, not in this PR.

Reply:
As discussed, we'll do this in a follow-up pull request.